packages/app/obojobo-repository/shared/components/notification.test.js

```diff
@@ -0,0 +1,151 @@
+import React from 'react'
+import { create, act } from 'react-test-renderer'
+import Notification from './notification'
+
+describe('Notification component', () => {
+	beforeAll(() => {
+		Object.defineProperty(document, 'cookie', {
+			value: '',
+			writable: true
+		})
+	})
+
+	test('renders without crashing', () => {
+		const component = create(<Notification onDataFromNotification={() => {}} />) // Provide a mock function for onDataFromNotification
+		const tree = component.toJSON()
+		expect(tree).toMatchSnapshot()
+	})
+
+	test('loads notifications from cookies on mount', () => {
+		const onDataFromNotification = jest.fn()
+		const notificationValue = [{ key: 1, text: 'Test Notification', title: 'Test Title' }]
+		document.cookie = `notifications=${JSON.stringify(notificationValue)}`
+
+		const component = create(<Notification onDataFromNotification={onDataFromNotification} />)
+		const tree = component.toJSON()
+
+		expect(tree).toMatchSnapshot()
+		expect(document.cookie).toBe(
+			`notifications=${encodeURIComponent(JSON.stringify(notificationValue))};`
+		)
+	})
+
+	test('handles click on exit button and updates state and cookie', () => {
+		const onDataFromNotification = jest.fn()
+		const notificationValue = [
+			{ key: 1, text: 'Notification1', title: 'Title1' },
+			{ key: 2, text: 'Notification2', title: 'Title2' }
+		]
+		document.cookie = `notifications=${JSON.stringify(notificationValue)}`
+
+		const reusableComponent = <Notification onDataFromNotification={onDataFromNotification} />
+		let component
+		act(() => {
+			component = create(reusableComponent)
+		})
+
+		let tree = component.toJSON()
+		expect(tree).toMatchSnapshot()
+
+		// Simulate click on exit button for the first notification
+		const key = 0
+		const exitButtons = component.root.findAllByProps({ className: 'notification-exit-button' })
+
+		if (exitButtons[key]) {
+			act(() => {
+				exitButtons[key].props.onClick()
+			})
+
+			tree = component.toJSON()
+			expect(tree).toMatchSnapshot()
+			expect(document.cookie).toBe(
+				`notifications=${encodeURIComponent(JSON.stringify(notificationValue))};`
+			)
+		}
+	})
+
+	test('hides the notification on exit button click', () => {
+		const onDataFromNotification = jest.fn()
+		document.cookie =
+			'notifications=' +
+			JSON.stringify([{ key: 1, text: 'Test Notification', title: 'Test Title' }])
+
+		const elementToExit = document.getElementsByClassName('notification-exit-button')[0]
+		const reusableComponent = <Notification onDataFromNotification={onDataFromNotification} />
+		let component
+		act(() => {
+			component = create(reusableComponent)
+		})
+
+		let tree = component.toJSON()
+		expect(tree).toMatchSnapshot()
+
+		const key = 0
+		const exitButtons = component.root.findAllByProps({ className: 'notification-exit-button' })
+
+		act(() => {
+			exitButtons[key].props.onClick()
+		})
+
+		tree = component.toJSON()
+		expect(tree).toMatchSnapshot()
+
+		if (elementToExit) {
```

Comment:
This is another case where you should use `expect` rather than `if`. Ideally you should populate `elementToExit` and then have an immediate `expect` to make sure it's what you expect it to be, at which point this `if` would be unnecessary because you've already made sure it exists.

Reply:
yes, removing the if worked perfectly. Seems it was redundant to check if it existed and then expect it to exist and be undefined in value